lib/spack/spack/test/conftest.py

```diff
@@ -301,8 +301,16 @@ def use_configuration(config):
     """Context manager to swap out the global Spack configuration."""
     saved = spack.config.config
     spack.config.config = config
+
+    # This is essential, otherwise the cache will create weird side effects
```

Comment:
This comment should be updated: for now it could be read to imply that this is the only case where caching causes errors. For example, the following would be acceptable:

> Avoid using real spack configuration that has been cached by other tests, and avoid polluting the cache with spack test configuration (including modified configuration)

Reply:
Done.